### PR TITLE
Add Exercise04

### PR DESCRIPTION
**Pull request by huynhngocminh2511**

### CODE REVIEW

File: Exercise04/Program.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+{
+    class Program
+    {
+        public static IEnumerable<string> GetAllFiles(string path, Func<string, bool> endWith)
+        {
+            var files = Directory.GetFiles(path, "*", SearchOption.AllDirectories);
+
+            var list = new List<string>();
+
+            foreach (var i in files)
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: Bad variable, it should be `file`

File: Exercise04/Program.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace Exercise04
+{
+    class Program
+    {
+        public static IEnumerable<string> GetAllFiles(string path, Func<string, bool> endWith)
+        {
+            var files = Directory.GetFiles(path, "*", SearchOption.AllDirectories);
+
+            var list = new List<string>();
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: `list` is unused

File: Exercise04/Program.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Exercise04
+{
+    class Program
+    {
+        public static IEnumerable<string> GetAllFiles(string path, Func<string, bool> endWith)
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: `endWith` should be renamed to `filter` or `condition`

File: Exercise04/Program.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                    yield return i;
+                }
+            }
+        }
+
+        static void Main(string[] args)
+        {
+            var pathFolder = @"C:\Users\huynh\OneDrive";
+
+            foreach(var i in GetAllFiles(pathFolder, x => x.EndsWith(".txt")))
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: How about `GetAllFiles(pathFolder, x => x.EndsWith(".txt")).ToList().ForEach(Console.WriteLine)`

File: Exercise04/Program.cs (hunk 3 above)
Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

File: Exercise04/Program.cs (hunk 2 above)
Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

File: Exercise04/Program.cs (hunk 1 above)
Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

File: Exercise04/Program.cs (hunk 4 above)
Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

**Pull request merged**